idom_router/routers/starlette.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+import re
+from typing import Any
+
+from starlette.convertors import Convertor
+from starlette.routing import compile_path
+
+from idom_router.types import Route
+from idom_router.core import create_router
+
+
+def compile_starlette_route(route: Route) -> StarletteRouteResolver:
+    pattern, _, converters = compile_path(route.path)
+    return StarletteRouteResolver(pattern, converters)
+
+
+class StarletteRouteResolver:
```

Comment:
Probably should pull this out of idom-router

Reply:
I've removed the dependency on starlette and just implemented a simple router under `idom_router.simple`.